src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -521,12 +521,30 @@ private void setFlsHeaders(EvaluatedDlsFlsConfig dlsFls, ActionRequest request)
                         }
                     }
                 } else {
-                    threadContext.putHeader(
-                        ConfigConstants.OPENDISTRO_SECURITY_MASKED_FIELD_HEADER,
-                        Base64Helper.serializeObject((Serializable) maskedFieldsMap)
-                    );
-                    if (log.isDebugEnabled()) {
-                        log.debug("attach masked fields info: {}", maskedFieldsMap);
+                    boolean shouldIncludeFlsHeaders = true;
+                    if (request instanceof SearchRequest) {
+                        SearchRequest searchRequest = (SearchRequest) request;
+                        boolean hasOnlyCountOrCardinality = true;
+                        if (searchRequest.source().aggregations() != null) {
+                            for (AggregationBuilder af : searchRequest.source().aggregations().getAggregatorFactories()) {
+                                if (!(af.getType().equals("cardinality") || af.getType().equals("count"))) {
```

Comment:
nit: Thoughts on replacing `cardinality` with [CardinalityAggregationBuilder.NAME](https://github.com/opensearch-project/OpenSearch/blob/e67ced73226453d5a5504c78f3b7d5ae90b4914e/server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregationBuilder.java#L65).

Tried to find one for `count` field but is defined in many different contexts: https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch+%22%3D+%5C%22count%5C%22%22&type=code

Reply:
also, should this be a configurable list? or is it exhaustive to these two entries?

Reply:
Meant to open this on my fork at first as a pre-draft stage for this >.<

Looks like the value should be `value_count`: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/aggregations/metrics/ValueCountAggregationBuilder.java#L60